### PR TITLE
chore: reduce FE e2e ci time

### LINKED ISSUE #1614: Optimized front-end e2e test time
# Feature request
Currently the front-end e2e test takes 30-40 minutes to run through the test, which takes too long, we need to do something to reduce the time
image

## Describe alternatives you've considered
 - [ ] Caching node_modules
 - [ ] Parallel running front-end e2e tests
 
refer link: https://docs.cypress.io/guides/continuous-integration/github-actions.html#Parallelization

### PR DESCRIPTION
**Pull request by LiteSun**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**
Currently, front-end e2e CI takes around 40+ minutes to run through. This PR is used to reduce the front-end e2e time by about 20mins.

**Related issues**

resolves #1614 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: .github/workflows/frontend-e2e-test.yml [hunk 1]
```diff
@@ -15,20 +15,18 @@ jobs:
...
-        image: bitnami/etcd:3.4.13
+        image: 'bitnami/etcd:3.4.13'
         ports:
-          - 2379:2379
-          - 2380:2380
+          - '2379:2379'
+          - '2380:2380'
         env:
-          ALLOW_NONE_AUTHENTICATION: yes
+          ALLOW_NONE_AUTHENTICATION: 'yes'
```
Comment (juzhiyuan) [inline .github/workflows/frontend-e2e-test.yml:25]: auto format by your vscode?

File: web/cypress.json [hunk 2]
```diff
@@ -2,5 +2,7 @@
   "viewportWidth": 1920,
   "viewportHeight": 1080,
   "baseUrl": "http://localhost:8000",
-  "video": true
+  "video": true,
+  "numTestsKeptInMemory": 0,
```
Comment (juzhiyuan) [inline web/cypress.json:6]: any resources about this field?

**Comment by juzhiyuan** [comment]
If this PR works and gets merged, it will save a lot of us life!!

File: web/cypress.json (hunk 2 above)
Comment (LiteSun) [reply web/cypress.json:6]: see: https://docs.cypress.io/guides/references/configuration#Options

**Comment by LiteSun** [comment]
There was a problem with the node_modules cache, Will fix it later.

File: .github/workflows/frontend-e2e-test.yml [hunk 3]
```diff
@@ -15,20 +17,18 @@ jobs:
...
```
Comment (moonming) [inline .github/workflows/frontend-e2e-test.yml:27]: Please don't change these lines.

File: .github/workflows/frontend-e2e-test.yml [hunk 4]
```diff
@@ -37,31 +37,54 @@ jobs:
       - name: Setup golang environment
         uses: actions/setup-go@v2.1.3
         with:
-          go-version: "1.13"
+          go-version: '1.13'
```
Comment (moonming) [inline .github/workflows/frontend-e2e-test.yml:40]: ditto

Comment (LiteSun) [reply .github/workflows/frontend-e2e-test.yml:40]: Got it.

Comment (LiteSun) [reply .github/workflows/frontend-e2e-test.yml:40]: The rest of the formatting has been changed. The official documentation for this place uses single quotes, so let's leave it as it is, probably having changed the formatting wrong before.

image

**Review by juzhiyuan — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Review by Yiyiyimu — APPROVED** [review]

**Comment by juzhiyuan** [comment]
ping

**Review by moonming — APPROVED** [review]

**Pull request merged**